src/value/generator.rs

```diff
@@ -98,95 +98,96 @@ pub trait BaseGenerator {
     #[inline(always)]
     fn write_string(&mut self, string: &str) -> io::Result<()> {
         stry!(self.write_char(b'"'));
-        let mut string = string.as_bytes();
-        let mut len = string.len();
-        let mut idx = 0;
+        let string = string.as_bytes();
+//        let mut string = string.as_bytes();
+//        let mut len = string.len();
+//        let mut idx = 0;
 
-        unsafe {
+//        unsafe {
```

Comment:
this should be commented in again. a simple solution would be to guard the unsafe with a 

```rust
#[cfg(target_feature = "avx2")]
unsafe {
...
}
```

That way we don't need compile time checks at all and the code only gets included when the right intrinsics are present.